File: ios/WeightLossBetting/Data/Repository/NotificationRepository.swift
import Foundation

class NotificationRepository: BaseRepository {
    private lazy var apiService = APIService.shared
    
    func registerDeviceToken(token: String) async -> NetworkResult<Void> {
        return await safeApiCall { completion in
            self.apiService.registerDeviceToken(token: token, completion: completion)
        }
    }
}

File: android/app/src/main/java/com/weightloss/betting/data/repository/NotificationRepository.kt
package com.weightloss.betting.data.repository

import com.weightloss.betting.data.remote.ApiService
import com.weightloss.betting.data.remote.NetworkResult
import javax.inject.Inject
import javax.inject.Singleton

/**
 * 通知 Repository - 处理通知相关的数据操作
 */
@Singleton
class NotificationRepository @Inject constructor(
    private val apiService: ApiService
) : BaseRepository() {
    
    /**
     * 注册设备令牌
     */
    suspend fun registerDeviceToken(token: String): NetworkResult<Unit> {
        return safeApiCall {
            apiService.registerDeviceToken(mapOf("token" to token, "platform" to "android"))
        }
    }
}
